WordPress/gutenberg
# `DateTimePicker`: fix date format when switching to 12-hour time format

Fixes: #17970


## Description

Updated the `TimePicker` component to show time in `MM-DD-YYYY` format as mentioned in the [docs](https://developer.wordpress.org/block-editor/reference-guides/components/date-time/#is12hour) 
## How has this been tested?



- Run storybook
- Navigate to `TimePicker` component
- Under the knob section select 12-hours format
- Verify the format
## Screenshots 
![image](https://user-images.githubusercontent.com/21127788/146429029-b484e1e9-8e0f-413b-99d3-8353fd0564c2.png)

## Types of changes




Bugfix
## Checklist:
